### DISCUSSION TITLE
Dynamic log package

### DISCUSSION THREAD

**Discussion by MM25Zamanian** [IBM/pwa-lit-template#501] [Ideas]
Hello. 
I have an idea to build a printing package that prints based on the configuration file settings and notifies the developer of the status of the application in development mode.

Forgive me if I am not fluent in English

**Comment by MM25Zamanian** [comment]
If you like this idea, I can do it

**Comment by abdonrd** [comment]
I don't quite understand what you mean; could you give more detail? Thanks!

**Reply by MM25Zamanian** [reply]
A developer must log in to the browser console to know the different modes of their web application, but this depends on the <development> and <production> modes.

I think the starter should meet this need because it has arranged the file structure itself

It's very simple
But this is a part that helps developers a lot
Of course, this is my idea and I have not seen it anywhere yet

**Reply by abdonrd** [reply]
We already show how to see the environment in the footer:

https://github.com/IBM/pwa-lit-template/blob/6f25fdedcb657c4c1351be3b348b0dade27dae32/src/components/app-index.ts#L97

I think it's enough.

**Reply by MM25Zamanian** [reply]
No, no, you do not understand what I mean.
See this:

```js
‍‍function _log(message: string) {
  if (config.environment === 'development') {
    console.log('🚀 ~  ~ message', message);
  }
}


_log("Message");
```

Print and display events to the developer's choice in the browser console
Of course, the above code is just a simple example to understand
The prototype is more professional and has more capabilities

**Reply by abdonrd** [reply]
Oh, got it! Thanks for the example! 👏🏻 

If we do, we should avoid shipping all that code in the production build.

But anyway, right now we're not interested in adding a logger to this template; we want to keep it as small as possible.

**Reply by MM25Zamanian** [reply]
thank you
My ideas help whenever you decide to make your starter bigger

**Comment by MM25Zamanian** [comment]
@abdonrd Please read [this file](https://github.com/MM25Zamanian/pwa-lit-starter/blob/develop/src/utils/base-element.ts)
This file is just an exercise
